### PR TITLE
Guests will be able to add their own location for in person meetings

### LINKED ISSUE #5226: [CAL-265] Guests should be able to add their own location for in-person meetings
### Is your proposal related to a problem?

(Write your answer here.)

One of the main issues that we're facing with [cal.com](http://cal.com) is when our guests want us to go to their office or warehouse.

### Describe the solution you'd like

(Describe your proposed solution here.)

When creating an event type, if in location you select ***In person***, you must specify the address. However, when selecting the option for phone number, [cal.com](http://cal.com) already allows the guest to specify the phone number. The location for in-person meetings should work the same as with the phone numbers.

### Describe alternatives you've considered

(Write your answer here.)

I tried the suggestion the support team gave, which is adding a new required field to the booking form for the guest to add the location while typing in their details. This workaround isn't good enough, as the functionality should be built in like with the phone numbers as location for the meeting.

### Additional context

(Write your answer here.)

Calendly has this functionality already, and [cal.com](http://cal.com) truly needs it.

> From [Linear-GitHub Sync](https://synclinear.com)

### PR DESCRIPTION
**Pull request by om-ray**
## What does this PR do?

Guests can now add their own locations for in-person meetings. 

Fixes #5226 and #5192

**Environment**: Staging(main branch) / Production

Loom: 
https://www.loom.com/share/ed59c0a7899441fab6ec60ebeee32b0b

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by om-ray** [comment]
@PeerRich @hariombalhara @zomars

**Comment by PeerRich** [comment]
can you add a loom video?

**Comment by om-ray** [comment]
@PeerRich I'll add one right now!

**Comment by PeerRich** [comment]
CleanShot 2022-10-31 at 23 09 19@2x
it says "attendeeInPerson" here in location and no the actual input

**Review by J4VMC — APPROVED** [review]
Delivers the requested functionality.

**Comment by om-ray** [comment]
> CleanShot 2022-10-31 at 23 09 19@2x
> 
> it says "attendeeInPerson" here in location and no the actual input

@PeerRich Sorry for the late response, I'm working on this right now!

**Comment by om-ray** [comment]
@PeerRich I've fixed the issue you mentioned and would love if you would review the PR one more time!

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -657,16 +663,39 @@ const BookingPage = ({
...
-                        id="phone"
+                        id={
+                          selectedLocationType === LocationType.Phone
+                            ? "phone"
+                            : selectedLocationType === LocationType.AttendeeInPerson
+                            ? "attendeeAddress"
+                            : ""
+                        }
                         required
                         disabled={disableInput}
```
Comment (CarinaWolli) [inline apps/web/components/booking/pages/BookingPage.tsx:700]: I think we need to remove this disabled here. And instead not show the address and phone number input at all when it is a reschedule

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Great job so far! I found some issues, that need to be fixed before I can approve. 

In addition to the comments, I found one more bug. In the following case I am not able to book the event with Google Meet as the location: 
Screenshot 2022-11-02 at 16 30 42

File: apps/web/components/booking/pages/BookingPage.tsx (hunk 1 above)
Comment (om-ray) [reply apps/web/components/booking/pages/BookingPage.tsx:700]: @CarinaWolli I think the problem with the google meet not going through is with the `bookingFormSchema` requiring the attendeeAddress field a minimum length of 1, I've fixed the problem on my end and that seemed to be the solution! Thank you for your review!

File: packages/emails/src/components/LocationInfo.tsx [hunk 2]
```diff
@@ -71,6 +71,18 @@ export function LocationInfo(props: { calEvent: CalendarEvent; t: TFunction }) {
         }
       />
     );
+  } else if (!isPhone) {
```
[the remark this answers is not in the document]
Comment (om-ray) [reply packages/emails/src/components/LocationInfo.tsx:74]: @CarinaWolli This is a change I made when I was trying to get the location to show on the booking page that I completely forgot to remove, thank you so much for telling me about this, I thought I had already removed all the unnecessary changes I had made! Thank you for your review!

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
I found one last thing. We should not show it when editing the location of a booking. So simply filter it out from the location options like we are doing it for the Attendee PhoneScreenshot 2022-11-03 at 13 31 05

**Comment by om-ray** [comment]
@CarinaWolli I'll fix this asap! Thank you for your review!

**Review by CarinaWolli — APPROVED** [review]
Looks good now 🙌🏻

**Pull request merged**